akka-stream/src/main/scala/akka/stream/javadsl/Tcp.scala

```diff
@@ -159,23 +170,32 @@ class Tcp(system: ExtendedActorSystem) extends akka.actor.Extension {
    *                  therefore it is the default setting.
    *                  If set to false, the connection will immediately closed once the client closes its write side,
    *                  independently whether the server is still attempting to write.
+   * @param keepOpenOnPeerClosed
+   *                  When the remote side has closed the incoming side of the connection, keep the outgoing side
+   *                  open to keep sending data until finished.
    */
-  def outgoingConnection(
-    remoteAddress:  InetSocketAddress,
-    localAddress:   Optional[InetSocketAddress],
-    options:        JIterable[SocketOption],
-    halfClose:      Boolean,
-    connectTimeout: Duration,
-    idleTimeout:    Duration): Flow[ByteString, ByteString, CompletionStage[OutgoingConnection]] =
-    Flow.fromGraph(delegate.outgoingConnection(remoteAddress, localAddress.asScala, immutableSeq(options), halfClose, connectTimeout, idleTimeout)
+  def connect(
+    remoteAddress:        InetSocketAddress,
+    localAddress:         Optional[InetSocketAddress],
+    options:              JIterable[SocketOption],
+    halfClose:            Boolean,
+    connectTimeout:       Duration,
+    idleTimeout:          Duration,
+    keepOpenOnPeerClosed: Boolean): Flow[ByteString, ByteString, CompletionStage[OutgoingConnection]] =
```

Comment:
if we are anyway changing this it might be worth creating a settings class for these options. Easier to evolve. options, halfClose, connectTimeout, idleTimeout, keepOpenOnPeerClosed could all go into that